### DISCUSSION TITLE
🐛 Select Initial Path on the navigation pane

### DISCUSSION THREAD

**Discussion by Harshul2023** [bdlukaa/fluent_ui#1148] [General]
```dart
class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    final appTheme = Get.put(AppThemeController());

    // Use Obx for reactive state management
    return FluentApp.router(
      title: appTitle,
      // themeMode: appTheme.mode,
      debugShowCheckedModeBanner: false,
      // // color: appTheme.color,
      // darkTheme: FluentThemeData(
      //   brightness: Brightness.dark,
      //   // accentColor: appTheme.color,
      //   visualDensity: VisualDensity.standard,
      //   focusTheme: FocusThemeData(
      //     glowFactor: is10footScreen(context) ? 2.0 : 0.0,
      //   ),
      // ),
      theme: FluentThemeData(
        navigationPaneTheme: NavigationPaneThemeData(
          tileColor: WidgetStateProperty.all(
              NAVIGATION_MENU_BACKGROUND.withOpacity(0.2)),

          labelPadding: EdgeInsets.symmetric(vertical: 15),
          highlightColor: SELCTED_NAVIGATION_COLOR,

          overlayBackgroundColor: Colors.transparent,
          backgroundColor: NAVIGATION_MENU_BACKGROUND,
          // Ensure background matches theme
          selectedIconColor:
              WidgetStateProperty.all(Color.fromRGBO(255, 255, 255, 0.8)),
          selectedTextStyle: WidgetStateProperty.all(TextStyle(
              color: Colors.white, fontWeight: FontWeight.w600, fontSize: 16)),
          unselectedTextStyle: WidgetStateProperty.all(TextStyle(
              color: Colors.white, fontWeight: FontWeight.w400, fontSize: 16)),
          unselectedIconColor: WidgetStateProperty.all(Colors.white),
          selectedTopTextStyle: WidgetStateProperty.all(TextStyle(
              color: Colors.red, fontWeight: FontWeight.w600, fontSize: 16)),
        ),
        brightness: Brightness.light,
        accentColor: Colors.blue,
        menuColor: Colors.black,
        scaffoldBackgroundColor: Colors.white,
      ),
      // locale: appTheme.locale,
      builder: (context, child) {
        return Obx(() {
          return Get.find<SideBarController>().isLogin.value
              ? Directionality(
            textDirection: appTheme.textDirection.value,
            child: NavigationPaneTheme(
              data: NavigationPaneThemeData(),
              child: child!,
            ),
          )
              : LoginScreen();
        });
        // return
        // return Container(color: Colors.red,);
      },
      routeInformationParser: router.routeInformationParser,
      routerDelegate: router.routerDelegate,
      routeInformationProvider: router.routeInformationProvider,
    );
  }
}
```

i am using the exmple application only
but depending upon the isLoginState i want to show login page else the MyApp Page
Is there a way i can decide on which path i want to go 
like from login page i want to go to buttons, or popups and that tab is opened directly instead of "/"

**Comment by bdlukaa** [comment]
You need to improve your state management solution. To set the default route of the `NavigationView`, just set the `selected` property to the desired value.

https://bdlukaa.github.io/fluent_ui/#/navigation/navigation_view

**Comment by Harshul2023** [comment]
when i am setting the selected the index as 1  the second pane is successfully selected but by default the home route is opening why so?
final router = GoRouter(navigatorKey: rootNavigatorKey,
    initialLocation: '/devices',
    routes: [
  ShellRoute(
    navigatorKey: _shellNavigatorKey,
    builder: (context, state, child) {
      return MyHomePage(
        shellContext: _shellNavigatorKey.currentContext,
        child: child,
      );
    },
    routes: <GoRoute>[
      /// Home
      GoRoute(path: '/', builder: (context, state) => const UserManagement()),

      /// Settings
      GoRoute(
          path: '/devices',
          builder: (context, state) => const DeviceManagement()),
      GoRoute(
          path: '/batches',
          builder: (context, state) => const BatchManagement()),
      GoRoute(
          path: '/firmwares',
          builder: (context, state) => const FirmwareManagement()),
      GoRoute(
          path: '/manage-delivery',
          builder: (context, state) => const DeliveryManagementTab()),
    ],
  ),
]);
if i also define my initial location every time i login i want that i can decide what should be the initial location 
howo to implement that

**Reply by bdlukaa** [reply]
The pane and the routes are two completely different things. You have to manage them separately.

**Comment by Harshul2023** [comment]
Hy thanks for your help I sincerely apologies for bothering you It was my mistake
The Solution was using shellroute sub navigation of the pages
sincerely thanks for your time and help
just one more help neede 
why is my dialog coming like this
<img width="380" alt="Screenshot 2024-11-29 at 8 28 37 PM" src="https://github.com/user-attachments/assets/d4513cb2-bd7d-4c8b-9bd3-2059ff605e8b">

i want dialog to be completely white background without bacground of page visible
 showDialog(
            context: context,
            builder: (context) {
              return material.AlertDialog(
                title: Text(
                  'Log Out',
                  style: figtreeTextStyle(
                    fontWeight: FontWeight.w800,
                    color: Colors.black,
                    fontSize: 23,
                  ),
                ),
                content: Text(
                  'Are you sure you want to log out?',
                  style: figtreeTextStyle(
                    fontWeight: FontWeight.w500,
                    color: SECONDARY_COLOR,
                    fontSize: 18,
                  ),
                ),
                actions: [
                  Row(
                    children: [
                      Expanded(
                        child: deliveryTabButtons(
                          "No",
                          buttonColor: Color(0xFFB2B2B2),
                              () {
                            Navigator.pop(context);
                          },
                        ),
                      ),
                      SizedBox(width: 10),
                      Expanded(
                        child: deliveryTabButtons(
                          "Yes",
                              () async {
                            
                      ),
                    ],
                  ),
                ],
              );
            },
          );

**Reply by bdlukaa** [reply]
use `material.showDialog` instead of just `showDialog`.
